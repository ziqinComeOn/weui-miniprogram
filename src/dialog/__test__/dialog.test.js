import path from 'path'
import simulate from 'miniprogram-simulate'

describe('dialog', () => {
    let id

    beforeAll(() => {
        id = simulate.load(path.resolve(__dirname, './index'), { less: true })
    })

    test('basic dialog', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))
        expect(container.toJSON()).toMatchSnapshot()
    })

    test('buttons', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))

        const dialog = container.querySelector('.dialog')
        const [cancelBtn, confirmBtn] = dialog.querySelectorAll('.weui-dialog__btn')

        cancelBtn.dispatchEvent('tap')
        await simulate.sleep(10)
        expect(container.data.buttonValue).toBe('取消')

        confirmBtn.dispatchEvent('tap')
        await simulate.sleep(10)
        expect(container.data.buttonValue).toBe('确定')
    })

    test('close', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))

        const dialog = container.querySelector('.dialog')

        dialog.querySelector('.weui-mask').dispatchEvent('tap')
        await simulate.sleep(10)
        expect(container.data.closed).toBe(true)
        expect(container.toJSON()).toMatchSnapshot()
    })
})
