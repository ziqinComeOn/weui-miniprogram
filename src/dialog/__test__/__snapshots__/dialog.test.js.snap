// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog basic dialog 1`] = `
<main>
  <mp-dialog
    class="dialog"
    bind:buttontap="buttontap"
    bind:close="close"
  >
    <wx-view
      class="weui-mask "
      bind:tap="close"
    />
    <wx-view
      class="weui-dialog__wrp "
      bind:tap="close"
    >
      <wx-view
        class="weui-dialog"
        catch:tap="stopEvent"
      >
        <wx-view
          class="weui-dialog__hd"
        >
          <wx-view
            class="weui-dialog__title"
          >
            dialog title
          
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-dialog__bd"
        />
        <wx-view
          class="weui-dialog__ft"
        >
          <wx-view
            class="weui-dialog__btn weui-dialog__btn_default "
            data-index="{{0}}"
            bind:tap="buttonTap"
          >
            取消
          </wx-view>
          <wx-view
            class="weui-dialog__btn weui-dialog__btn_primary "
            data-index="{{1}}"
            bind:tap="buttonTap"
          >
            确定
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
  </mp-dialog>
</main>
`;

exports[`dialog close 1`] = `
<main>
  <mp-dialog
    class="dialog"
    bind:buttontap="buttontap"
    bind:close="close"
  >
    <wx-view
      class="weui-mask weui-mask_hidden"
      bind:tap="close"
    />
  </mp-dialog>
</main>
`;
